src/components/LandingPageComponent.jsx

```diff
@@ -347,85 +207,46 @@ class LandingPageComponent extends Component {
                   Cancel Transfer
                 </Button>
               </Grid>
-            }
+            )}
           </Grid>
         </ExpansionPanelDetails>
       </ExpansionPanel>
     )
   }
 
-  renderWalletSection = (props) => {
-    const { classes, actionsPending, cloudWallet, walletBalanceCurrencyAmount } = this.props
+  renderWalletSection = props => {
+    const { classes, push } = this.props
+    const { walletSelection } = this.state
     // Do not remove <div style={{ padding: 10 }}>
     // See https://material-ui.com/components/grid/#limitations
     return (
-      <Grid container alignItems='center' justify='center' className={classes.coloredBackgrond} >
+      <Grid container alignItems='center' justify='center' className={classes.coloredBackgrond}>
         <Container maxWidth='lg'>
           <Box mb={2}>
-            <Typography variant='h2'>Drive Wallet Balance</Typography>
+            <Typography variant='h2'>Transfer From</Typography>
           </Box>
-
-          {/* <div style={{ padding: '0 10px 0 10px' }}> */}
           <Grid container direction='row' alignItems='center' spacing={2}>
-            {walletCryptoSupports['drive'].map((c, i) => {
-              c = getCrypto(c.cryptoType)
+            {walletSelections.map((w, i) => {
               return (
-                <Grid item sm={12 / walletCryptoSupports['drive'].length} key={i} xs={12}>
-                  <Card>
-                    <SpotlightTarget
-                      name={c.cryptoType === 'ethereum' ? 'two' : undefined}
-                    >
-                      {cloudWallet.crypto[c.cryptoType] &&
-                      !actionsPending.getCloudWallet
-                        ? <>
-                          <Grid container direction='row' alignItems='center' justify='center'>
-                            <Typography variant='h1' align='center'>
-                              {numeral(
-                                utils.toHumanReadableUnit(
-                                  cloudWallet.crypto[c.cryptoType][0].balance,
-                                  getCryptoDecimals(c.cryptoType)
-                                )
-                              ).format('0.000a')}
-                            </Typography>
-                            <Typography align='right' className={classes.balanceCurrencyText}>
-                          (≈ {walletBalanceCurrencyAmount[c.cryptoType]})
-                            </Typography>
-                          </Grid>
-                        </>
-                        : <Box display='flex' justifyContent='center'><CircularProgress color='primary' /></Box>
-                      }
-                      <Typography variant='body2' align='center'>{c.symbol}</Typography>
-                    </SpotlightTarget>
-                  </Card>
+                <Grid item sm={3} key={i} xs={12}>
+                  <WalletSelectionButton
```

Comment:
See suggestions beneath. Also, note that due to not using a single wallet selection component, it is easy to pass inconsistent props, e.g. from line 233~236, compared to the usages in other places.